physionet-django/project/models.py

```diff
@@ -1,2176 +1,11 @@
-    resource_type = models.PositiveSmallIntegerField(default=0)
-    publish_date = models.DateField()
-
-    # In case we want a citation
-    citation = models.CharField(blank=True, default='', max_length=1000)
-    citation_url = models.URLField(blank=True, default='')
-
-    contact_name = models.CharField(max_length=120, default='PhysioNet Support')
-    contact_affiliations = models.CharField(max_length=150, default='MIT')
-    contact_email = models.EmailField(max_length=255, default='webmaster@physionet.org')
-
-    # Put the references as part of the full description
-
-    def __str__(self):
-        return ' --- '.join([self.slug, self.title])
-
-    def publish(self, make_file_roots=False):
-        """
-        Turn into a published project
-        """
-        p = PublishedProject.objects.create(title=self.title,
-            doi=self.doi, slug=self.slug,
-            resource_type=ProjectType.objects.get(id=self.resource_type),
-            core_project=CoreProject.objects.create(),
-            abstract=self.abstract,
-            is_legacy=True, full_description=self.full_description,
-            version=self.version,
-            license=License.objects.get(name='Open Data Commons Attribution License v1.0')
-        )
-
-        # Have to set publish_datetime here due to auto_now_add of object
-        dt = datetime.combine(self.publish_date, datetime.min.time())
-        dt = pytz.timezone(timezone.get_default_timezone_name()).localize(dt)
-        p.publish_datetime = dt
-        p.save()
-
-        # Related objects
-        if self.citation:
-            PublishedPublication.objects.create(citation=self.citation,
-                url=self.citation_url, project=p)
-
-        Contact.objects.create(name=self.contact_name,
-            affiliations=self.contact_affiliations, email=self.contact_email,
-            project=p)
-
-        if make_file_roots:
-            os.mkdir(p.project_file_root())
-            os.mkdir(p.file_root())
-
-
-class GCP(models.Model):
-    """
-    Store all of the Google Cloud information with a relation to a project.
-    """
-    project = models.OneToOneField('project.PublishedProject', related_name='gcp',
-        on_delete=models.CASCADE)
-    bucket_name = models.CharField(max_length=150, null=True)
-    access_group = models.CharField(max_length=170, null=True)
-    is_private = models.BooleanField(default=False)
-    sent_zip = models.BooleanField(default=False)
-    sent_files = models.BooleanField(default=False)
-    managed_by = models.ForeignKey('user.User', related_name='gcp_manager',
-        on_delete=models.CASCADE)
-    creation_datetime = models.DateTimeField(auto_now_add=True)
-    finished_datetime = models.DateTimeField(null=True)
+from project.modelcomponents.legacy import *
+from project.modelcomponents.metadata import *
+from project.modelcomponents.publishedproject import *
+from project.modelcomponents.submission import *
+from project.modelcomponents.storage import *
```

Comment:
This module now has 11 imports from `modelcomponents`. There are 13 modules in that folder, but `unpublishedproject.py` and `generic.py` contain only abstract models.